src/parquet/schema/types.cc

```diff
@@ -280,6 +288,35 @@ std::unique_ptr<Node> PrimitiveNode::FromParquet(const void* opaque_element,
   return std::unique_ptr<Node>(result.release());
 }
 
+void GroupNode::ToParquet(void* opaque_element) const {
+  format::SchemaElement* element =
+    static_cast<format::SchemaElement*>(opaque_element);
+  element->__set_name(name_);
+  element->__set_num_children(field_count());
+  element->__set_repetition_type(ToThrift(repetition_));
+  if (logical_type_ != LogicalType::NONE) {
+    element->__set_converted_type(ToThrift(logical_type_));
+  }
+  // FIXME: SchemaFlattener does this for us: element->__set_field_id(id_);
+}
+
+void PrimitiveNode::ToParquet(void* opaque_element) const {
+  format::SchemaElement* element =
+    static_cast<format::SchemaElement*>(opaque_element);
+
+  element->__set_name(name_);
+  element->__set_num_children(0);
+  element->__set_repetition_type(ToThrift(repetition_));
+  if (logical_type_ != LogicalType::NONE) {
+    element->__set_converted_type(ToThrift(logical_type_));
+  }
+  element->__set_type(ToThrift(physical_type_));
+  // FIXME: SchemaFlattener does this for us: element->__set_field_id(id_);
+  element->__set_type_length(type_length_);
+  element->__set_precision(decimal_metadata_.precision);
+  element->__set_scale(decimal_metadata_.scale);
```

Comment:
`parquet-mr` does set the decimal metadata independent of the logical type. But in the case of parquet-mr, `decimal_metadata_` can be `null`. In these cases precision and scale are not set on the Thrift object. (See `parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java:122` )